<class 'json.decoder.JSONDecodeError'>: Expecting value: line 1 column 1 (char 0)

I got these error in this PR: https://github.com/All-Hands-AI/OpenHands/actions/runs/11724947894/job/32660052579

```
14:29:12 - openhands:INFO: resolve_issue.py:148 - ------------------------------
14:29:12 - openhands:INFO: resolve_issue.py:149 - END Runtime Completion Fn
14:29:12 - openhands:INFO: resolve_issue.py:150 - ------------------------------
14:29:12 - openhands:INFO: resolve_issue.py:233 - Got git diff for instance 4822:
--------
diff --git a/openhands/llm/llm.py b/openhands/llm/llm.py
index db5870c5..29a8ef68 100644
--- a/openhands/llm/llm.py
+++ b/openhands/llm/llm.py
@@ -62,6 +62,7 @@ FUNCTION_CALLING_SUPPORTED_MODELS = [
     'gpt-4o',
     'gpt-4o-mini',
     'claude-3.5-sonnet',  # OpenRouter model name
+    'claude-3.5-haiku',   # OpenRouter model name
 ]
--------
14:29:16 - openhands:INFO: resolve_issue.py:420 - Finished.
ERROR:root:  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/openhands_resolver/resolve_issue.py", line 568, in <module>
    main()
  File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/openhands_resolver/resolve_issue.py", line 549, in main
    asyncio.run(
  File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/asyncio/runners.py", line 194, in run
    return runner.run(main)
           ^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/asyncio/runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/asyncio/base_events.py", line 687, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/openhands_resolver/resolve_issue.py", line 403, in resolve_issue
    output = await process_issue(
             ^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/openhands_resolver/resolve_issue.py", line 253, in process_issue
    for success_indicator, explanation in zip(comment_success, json.loads(success_explanation)):
                                                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None

ERROR:root:<class 'json.decoder.JSONDecodeError'>: Expecting value: line 1 column 1 (char 0)
Error: Process completed with exit code 1.
```

Maybe we should add an exception handling in `json.loads(success_explanation)` in line 253 of `resolve_issue.py`

--- a/openhands_resolver/resolve_issue.py
+++ b/openhands_resolver/resolve_issue.py
@@ -250,10 +250,16 @@ async def process_issue(
 
         if issue_handler.issue_type == "pr" and comment_success:
             success_log = "I have updated the PR and resolved some of the issues that were cited in the pull request review. Specifically, I identified the following revision requests, and all the ones that I think I successfully resolved are checked off. All the unchecked ones I was not able to resolve, so manual intervention may be required:\n"
-            for success_indicator, explanation in zip(comment_success, json.loads(success_explanation)):
-                    status = colored("[X]", "red") if success_indicator else colored("[ ]", "red")
-                    bullet_point = colored("-", "yellow")
-                    success_log += f"\n{bullet_point} {status}: {explanation}"
+            try:
+                explanations = json.loads(success_explanation)
+            except json.JSONDecodeError:
+                logger.error(f"Failed to parse success_explanation as JSON: {success_explanation}")
+                explanations = [str(success_explanation)]  # Use raw string as fallback
+            
+            for success_indicator, explanation in zip(comment_success, explanations):
+                status = colored("[X]", "red") if success_indicator else colored("[ ]", "red")
+                bullet_point = colored("-", "yellow")
+                success_log += f"\n{bullet_point} {status}: {explanation}"
             logger.info(success_log)
         last_error = state.last_error if state.last_error else None
 

--- a/tests/test_resolve_issues.py
+++ b/tests/test_resolve_issues.py
@@ -300,6 +300,20 @@ async def test_process_issue(mock_output_dir, mock_prompt_template):
             "expected_success": False,
             "expected_error": "Agent failed to run or crashed",
             "expected_explanation": "Agent failed to run"
+        },
+        {
+            "name": "json_decode_error",
+            "run_controller_return": MagicMock(
+                history=[NullObservation(content="")],
+                metrics=MagicMock(get=MagicMock(return_value={"test_result": "passed"})),
+                last_error=None,
+            ),
+            "run_controller_raises": None,
+            "expected_success": True,
+            "expected_error": None,
+            "expected_explanation": "Non-JSON explanation",
+            "is_pr": True,
+            "comment_success": [True, False]  # To trigger the PR success logging code path
         }
     ]
 
@@ -320,9 +334,13 @@ async def test_process_issue(mock_output_dir, mock_prompt_template):
             mock_run_controller.side_effect = None
         
         mock_complete_runtime.return_value = {"git_patch": "test patch"}
-        handler_instance.guess_success.return_value = (test_case["expected_success"], None, test_case["expected_explanation"])
+        handler_instance.guess_success.return_value = (
+            test_case["expected_success"], 
+            test_case.get("comment_success", None), 
+            test_case["expected_explanation"]
+        )
         handler_instance.get_instruction.return_value = ("Test instruction", [])
-        handler_instance.issue_type = "issue"
+        handler_instance.issue_type = "pr" if test_case.get("is_pr", False) else "issue"
 
         with patch(
             "openhands_resolver.resolve_issue.create_runtime", mock_create_runtime
@@ -350,7 +368,8 @@ async def test_process_issue(mock_output_dir, mock_prompt_template):
             )
 
             # Assert the result
-            assert handler_instance.issue_type == "issue"
+            expected_issue_type = "pr" if test_case.get("is_pr", False) else "issue"
+            assert handler_instance.issue_type == expected_issue_type
             assert isinstance(result, ResolverOutput)
             assert result.issue == issue
             assert result.base_commit == base_commit

